core/src/main/java/fj/Ord.java

```diff
@@ -190,151 +170,91 @@ public A f(final A a, final A a1) {
    * An order instance for the <code>boolean</code> type.
    */
   public static final Ord<Boolean> booleanOrd = new Ord<Boolean>(
-      new F<Boolean, F<Boolean, Ordering>>() {
-        public F<Boolean, Ordering> f(final Boolean a1) {
-          return new F<Boolean, Ordering>() {
-            public Ordering f(final Boolean a2) {
-              final int x = a1.compareTo(a2);
-              return x < 0 ? Ordering.LT : x == 0 ? Ordering.EQ : Ordering.GT;
-            }
-          };
-        }
-      });
+          a1 -> a2 -> {
+            final int x = a1.compareTo(a2);
+            return x < 0 ? Ordering.LT : x == 0 ? Ordering.EQ : Ordering.GT;
+          });
```

Comment:
Use `Ord.ord(F<A, A<, Ordering>>)` to avoid specifying the type.  Applies for other changes to this file.